src/Microsoft.Health.Fhir.Shared.Client/FhirException.cs

```diff
@@ -46,14 +47,39 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        private string GetOperationId()
+        private string FormatMessage()
+        {
+            StringBuilder message = new StringBuilder();
+
+            string diagnostic = OperationOutcome?.Issue?.FirstOrDefault()?.Diagnostics;
+            string operationId = GetActivityId();
+
+            message.Append(StatusCode);
+            if (!string.IsNullOrWhiteSpace(diagnostic))
+            {
+                message.Append(": ").Append(diagnostic);
+            }
+
+            message.Append(" (").Append(operationId).AppendLine(")");
+
+            message.AppendLine("==============================================");
+            message.Append("Url: (").Append(Response.Response.RequestMessage?.Method.Method ?? "NO_HTTP_METHOD_AVAILABLE").Append(") ").AppendLine(Response.Response.RequestMessage?.RequestUri.ToString() ?? "NO_URI_AVAILABLE");
```

Comment:
Do we know that the URI is going to have the query values censored out if retrieved this way?

Reply:
That's a good question. 
When I was writing down these changes and checked where FhirException is being used. 
Currently, it's only used in test projects. We do not use it on FHIR server side. 
For this reason, I was thinking that the data exposed in the URI would be only with data from our tests. 
That was my train of thought. 

But, if you think that's a risk, I can change this part of the code to expose only the domain. 
That would be good enough.

Reply:
Ah, ok. I had this confused with the FhirException class in Core: https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.Core/Exceptions/FhirException.cs
We may want to think about having two unrelated classes with the same name.